### PR TITLE
Add Stateful Components in Elm.

### PR DESCRIPTION
**Pull request by n1k0**
A short article about stateful component API design in Elm.

### CODE REVIEW

File: _posts/2019-07-16-stateful-components-in-elm.md [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+
+![app demo](https://i.imgur.com/4Hs0IGa.gif)
+
+Many component/widget-oriented Elm packages feature a rather raw [TEA] API, namely exposing `Model`, `Msg(..)`, `init`, `update` and `view`, so you can basically import what defines an actual application and embed it within your own application.   
+
+![funny meme](https://i.imgur.com/aBXjI9N.png)
+
+With these, you usually end up writing things like this:
+
+```haskell
```
Comment (fbentz) [inline _posts/2019-07-16-stateful-components-in-elm.md]: Don't we have elm?

Comment (fbentz) [reply _posts/2019-07-16-stateful-components-in-elm.md]: Ok no, this is your last commit :)

File: _posts/2019-07-16-stateful-components-in-elm.md [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+
+- You need to `Cmd.map` and `Html.map` here and there
+- You need to pattern match `Widget.Msg` to intercept whatever event interests you...
+- ... meaning `Widget` exposes all `Msg`s, which are **implementation details** you now rely on.
+
+There's another way, which [Evan] explained in his now deprecated [elm-sortable-table] package. Among the many good points he has, one idea stroke me as brilliantly simple yet effective to simplify such stateful view components API design:
+
+> **State updates can be managed right from event handlers!**
+
+Let's imagine a simple counter; what if when clicking the *increment* button, instead of calling `onClick` with some `Increment` message, we would call **a user-provided one** with the new counter state updated accordingly?
```
Comment (vjousse) [inline _posts/2019-07-16-stateful-components-in-elm.md]: **a user-provided one** what? You mean **a user-provided function**?

Comment (n1k0) [reply _posts/2019-07-16-stateful-components-in-elm.md]: A user-provided message. To me it's rather clear, but for other readers I don't know. How would you rephrase it?

**Pull request merged**